src/detectors/ScanningStrategyDetector.ts

```diff
@@ -51,6 +52,7 @@ export class ScanningStrategyDetector implements IDetector<string, ScanningStrat
         accessType = await this.determineRemoteAccessType({ remoteUrl: path, serviceType });
       }
     } else {
+      this.repositoryConfig.serviceType = serviceType;
```

Comment:
Why you override `this.repositoryConfig.serviceType`? We need to return correct `serviceType` from the `detect()` function. That's not correct. It's absolutely unrelated to mutate the `repositoryConfig` here. It would be better to use returned value as default or to override `repositoryConfig` outside of the function.